controllers/virtinkcluster_controller.go

```diff
@@ -67,65 +71,136 @@ func (r *VirtinkClusterReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 }
 
 func (r *VirtinkClusterReconciler) reconcile(ctx context.Context, cluster *infrastructurev1beta1.VirtinkCluster) error {
-	if !cluster.DeletionTimestamp.IsZero() {
+	if !controllerutil.ContainsFinalizer(cluster, finalizer) {
```

Comment:
We should only add finalizer when object is not deleted. See https://book.kubebuilder.io/reference/using-finalizers.html